### PR TITLE
New layout for project and series detail pages

### PR DESCRIPTION
**Pull request by bonzini**
The next step... This will almost certainly require more changes, for example to avoid duplication in the templates (though this is pre-existing and due to differences between e.g. `project.extra_status` or `series.extra_status`; perhaps CBV can improve it by matching template inheritance with view inheritance?), but I'm posting it to show the first part of the work and perhaps merge a subset?

This series does not yet split each patch to an individual page. It only focuses on cleaning up the "ops" box and adding the status boxes.

### CODE REVIEW

**Comment by famz** [comment]
Any idea how to make the "Commits" panel display commits in order in this PR page? Currently I'm seeing this which makes no sense:


pr

**Comment by bonzini** [comment]
GitHub sorts commits by author-date, not commit-date. Looks like a `git rebase origin/master --exec 'git commit --amend --reset-author -C HEAD'` has done the trick.

File: mods/testing.py [hunk 1]
```diff
@@ -296,9 +301,7 @@ def prepare_project_hook(self, request, project):
                                    "class": "info",
                                    "content_html": self.build_config_html(request,
                                                                           project)})
```
Comment (famz) [inline mods/testing.py]: We could use format_html in self.build_config_html() and drop the "safe" filter in the template later, right?

File: mods/git.py [hunk 2]
```diff
@@ -105,29 +107,38 @@ def prepare_message_hook(self, request, message, detailed):
         l = message.get_property("git.apply-log")
         if l:
             failed = message.get_property("git.apply-failed")
-            message.extra_info.append({"title": "Git apply log",
-                                       "class": 'danger' if failed else 'default',
-                                       "content": l})
-            git_url = message.get_property("git.url")
+            git_url = urlquote(message.get_property("git.url"))
```
[the remark this answers is not in the document]
Comment (bonzini) [reply mods/git.py]: I'll remove that and look at fixing the database with a migration.

File: mods/testing.py (hunk 1 above)
Comment (bonzini) [reply mods/testing.py]: Yes, exactly.

**Pull request merged**